clang/lib/CodeGen/CGExpr.cpp

```diff
@@ -5842,7 +5842,7 @@ void CodeGenFunction::SetDivFPAccuracy(llvm::Value *Val) {
   if (!EltTy->isFloatTy())
     return;
 
-  if ((getLangOpts().OpenCL &&
+  if (((getLangOpts().OpenCL || getLangOpts().SYCLIsDevice) &&
```

Comment:
I'm actually surprised that we accept -cl-fp32-correctly-rounded-divide-sqrt. I don't think we'd want that to be the canonical way to do this because it implies an OpenCL backend, which isn't guaranteed. 

I'm a bit fuzzy as to what the intended accuracy semantics are in SYCL 2020. My understanding is that for math library functions, the accuracy is implementation defined but if an OpenCL backend is used the OpenCL accuracy requirements apply. I don't know if the same rules apply to single-precision division. Probably so?

That would complicate things, and I don't like it. There's a related email discussion happening right now. @gmlueck just added me to the distribution list. I'll add you, though I'm guessing you're already working with someone on the list.

Reply:
Thanks Greg, I think we had an internal misunderstanding about the OpenCL standard implications on SYCL device code. I'll move this to a draft.